### ISSUE TITLE
Resolve sdk, module-types, and settings/runner for resolve sources

### ISSUE THREAD

**Issue by Welchd1** [Welchd1/resolve-intellij-plugin#7]
Going to take a crack at adding both an sdk for the resolve compiler as well as module types and automatically generated run configurations which will allow us to execute `.resolve` files within the editor and see output within the builtin terminal. Consult the "go language" plugin for an example/target of this initial effort (it's likely that our version will be considerably less sophisticated starting out though).

**Comment by Welchd1** [comment]
Managed to add run configuration type that allows us to run  the compiler itself. This is all well and good, but not terribly useful for the users... So I decided to refocus this more towards 'running' executable RESOLVE files only (those containing a 'main' operation). To get this working, I need to create run config that executes translated code produced by our jar (rather than just invoking the jar as it currently does).

Apparently, this is can be accomplished relatively easily with a before run task. I'm still trying to grok all of things I need to do to get this working. In the meantime, see the questions I'm posting at the jetbrains forum: https://devnet.jetbrains.com/thread/466137?tstart=0
